beefy-gadget/src/worker.rs

```diff
@@ -109,20 +109,25 @@ impl<B, P> GossipValidator<B> for BeefyGossipValidator<B, P>
 where
 	B: Block,
 	P: Pair,
-	P::Public: Decode,
-	P::Signature: Decode,
+	P::Public: Debug + Decode,
+	P::Signature: Debug + Decode,
 {
 	fn validate(
 		&self,
 		_context: &mut dyn GossipValidatorContext<B>,
-		_sender: &sc_network::PeerId,
+		sender: &sc_network::PeerId,
 		mut data: &[u8],
 	) -> GossipValidationResult<B::Hash> {
-		if VoteMessage::<MmrRootHash, NumberFor<B>, P::Public, P::Signature>::decode(&mut data).is_ok() {
-			GossipValidationResult::ProcessAndKeep(self.topic)
-		} else {
-			GossipValidationResult::Discard
+		if let Ok(msg) = VoteMessage::<MmrRootHash, NumberFor<B>, P::Public, P::Signature>::decode(&mut data) {
+			if P::verify(&msg.signature, &msg.commitment.encode(), &msg.id) {
+				return GossipValidationResult::ProcessAndKeep(self.topic);
```

Comment:
I guess that's good enough, but long-term we should also make sure that `msg.id` is part of the declared validator set id. That needs persistence of previous validator sets though (#17).